### ISSUE TITLE
🚀 Feature: Enhancements to Share options

### ISSUE THREAD

**Issue by JNR8** [stonith404/pingvin-share#81]
### 🔖 Feature description

1: Ability to edit share options after creation, allowing for the expiry date to be changed, password changed, and email recipients to be changed.  
2: Ability to restrict each share to selected registered users (including auto email notification)
3: Ability to edit share to add/remove registered users (additional users added to the share should reive a new email, existing should not)
4: Adding non-registered users should provide a registration link to the provided email address. 
5: Ability for share creator to be notified when files in a share have been download (in the context above by a registered user).  

### 🎤 Pitch

An expansion on what you have already put in place.  you may already be thinking along these lines, but I did not notice anything in issues pane.

Being able to edit existing options would save time and storage space, as it stand currently when share expires or you want to add someone else to access the share the file has to be reuploaded and anew share used ( i know i send the link myself to someone but being able to amend the share would be a cleaner solution).  

In addition to that, restricting access to registered users only would allow a more granular level of control.  this could be expanded to log who had downloaded from a share, and also prevent their doing so again whilst allow other still listed to download from that share until the share expires.

**Comment by jennec** [comment]
Complete understand that much of this is a lot of work to do and may require a considerable amount of changes.   Obviously no pressure :)

**Comment by mgesquinosinx** [comment]
I find the request for "editing an existing link" very relevant and useful

**Comment by mp5family** [comment]
yes, would like to make a share link, maybe add it for 1 day or so to test the link, then able to change the date to Never. Or even go from Never to exp in a few days or so.
